apps/prairielearn/src/pages/administratorCourses/administratorCourses.js

```diff
@@ -19,8 +27,41 @@ router.get(
   '/',
   asyncHandler(async (req, res) => {
     res.locals.coursesRoot = config.coursesRoot;
-    const result = await sqldb.queryOneRowAsync(sql.select, []);
-    _.assign(res.locals, result.rows[0]);
+    const result = await sqldb.queryRow(
+      sql.select,
+      [],
+      z.object({
+        institutions: z.array(InstitutionSchema),
+        courses: z.array(
+          CourseSchema.extend({
+            institution: InstitutionSchema,
+          }),
+        ),
+        course_requests: z.array(
```

Comment:
The queries/types in each of these files look to be very similar. Is there any possibility of deduplication, e.g. moving the selection of the course requests into a new `models/course-request.ts` file?

Reply:
I have now created a new `models/course-request.ts` file  and `models/course-request.sql` and moved some of the queries/types to those files.